src/Aspire.Hosting/Extensions/ExecutableResourceBuilderExtensions.cs

```diff
@@ -40,6 +40,22 @@ public static IResourceBuilder<ExecutableResource> AddExecutable(this IDistribut
         return builder.WithManifestPublishingCallback(context => WriteExecutableAsDockerfileResource(context, builder.Resource));
     }
 
+    /// <summary>
+    /// Adds the arguments to be passed to an executable resource when the executable is started.
+    /// </summary>
+    /// <typeparam name="T">The resource type.</typeparam>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="args">The arguments to be passed to the executable when it is started.</param>
+    /// <returns>The <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<T> WithArgs<T>(this IResourceBuilder<T> builder, params string[] args) where T : ExecutableResource
```

Comment:
I copied this from https://github.com/dotnet/aspire/pull/1363 to be able to produce the bug and write test for it.